File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        nums1[:] = nums1[:m]
        
        if m == 0:
            nums1[:] = nums2
            return
        
        i, j, k = 0, 0, 0
        while i < m and j < n:
            if nums1[i] <= nums2[j]:
                i += 1
            else:
                nums1.insert(i, nums2[j])
                j += 1
                m += 1
        
        nums1.extend(nums2[j:])

File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if (nums2.size() == 0) return;

        vector<int> res(m + n, 0);
        int i = 0, j = 0, top = 0;

        while (i < m && j < n){
            if (nums1[i] == nums2[j]){
                res[top++] = nums1[i++];
                res[top++] = nums2[j++];
            }else if (nums1[i] < nums2[j]){
                res[top++] = nums1[i++];
            }else{
                res[top++] = nums2[j++];
            }
        }

        while (i < m){
            res[top++] = nums1[i++];
        }

        while (j < n){
            res[top++] = nums2[j++];
        }

        nums1 = res;
    }
};